File: SortingandSearching/02_apartments/02_Apartments.cpp
#include<bits/stdc++.h>
using namespace std;


int main()
{
    int n,k,x;cin>>n>>k>>x;
    vector<long long>desired(n);
    vector<long long>given(k);
    for(int i = 0; i < n; i++) 
        cin >> desired[i];
    for(int i = 0; i < k; i++) 
        cin >> given[i];
    
    sort(desired.begin(),desired.end());
    sort(given.begin(),given.end());

    int i=0,j=0,c =0;
    while(i<n && j<k)
    {
        if(abs(desired[i]-given[j])<=x)
        {
            c++;i++;j++;
        }
        else if(desired[i]<given[j])
            i++;
        else
            j++;
    }
    cout<<c;

    return 0;

}

File: SortingandSearching/02_apartments/02_Apartments.java
import java.util.*;
class Apartments
{
    public static void main(String[] args) 
    {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int k = sc.nextInt();
        int x = sc.nextInt();

        List<Long>desired = new ArrayList<>();
        List<Long>given = new ArrayList<>();

        for(int i=0;i<n;i++)
            desired.add(sc.nextLong());
        for(int i=0;i<k;i++)
            given.add(sc.nextLong());

        Collections.sort(desired);
        Collections.sort(given);
        sc.close();
        int i=0,j=0,c=0;
        while(i<n && j<k)
        {
            if(Math.abs(desired.get(i)-given.get(j))<=x)
            {
                i++;j++;c++;
            }
            else if(desired.get(i)<given.get(j))
                i++;
            else
                j++;
        }
        System.out.println(c);


        
    }
}